### PR TITLE
reverse conditions to aid readability

### PR DESCRIPTION
**Pull request by igalic**
by doing this, we separate the special case (throw) from the code
(return). This aids the reading flow, by reducing lots of nested ifs.

Furthermore, the special case is handled at the top (return/break early
principle). This also gives a feeling of "contracts" to the functions.

### CODE REVIEW

File: src/evaluate.js [hunk 1]
```diff
@@ -56,28 +56,30 @@ const table = {
...
     }
+    const args = node
+      .items
+      .slice(1)
+      .map(x => EVAL(stack, scope, x));
+    if (f.type === 'JSFunction') {
+      const args2 = [stack, scope].concat(args);
+      return f.f.apply(null, args2);
+    }
+    if (f.type === 'Function') {
```
Comment (wavebeem) [inline src/evaluate.js]: I would either keep the `else if` here or switch to a dispatch table. I've had to deal with too many long functions with `if` and no `else` and it's not clear if there's even a `return` down each branch.

**Comment by wavebeem** [comment]
Just one comment, otherwise looks good.

File: src/evaluate.js (hunk 1 above)
Comment (igalic) [reply src/evaluate.js]: i'm a big fan of dispatch, or … well, match, but we only have switch in js :(

Comment (wavebeem) [reply src/evaluate.js]: Yeah, I'm not a big fan of `switch`, so I definitely opt for dispatch tables when I can.

**Pull request merged**